source/code/plugins/changetracking_lib.rb

```diff
@@ -0,0 +1,134 @@
+require 'rexml/document'
+require 'cgi'
+require 'digest'
+require 'set'
+
+require_relative 'oms_common'
+
+class ChangeTracking
+
+    @@log = nil
+    def self.log= (value)
+        @@log = value
+    end
+
+    @@prev_hash = ""
+    def self.prev_hash= (value)
+        @@prev_hash = value
+    end
+
+    def self.instanceXMLtoHash(instanceXML)
+        ret = {}
+        propertyXPath = "PROPERTY"
+        instanceXML.elements.each(propertyXPath) { |inst| 
+            name = inst.attributes['NAME']
+            rexmlText = REXML::XPath.first(inst, 'VALUE').get_text # TODO escape unicode chars like "&amp;"
```

Comment:
I unescape the initial xml but the text stays escaped...
The behavior now is to send the escaped character up.
There are only a few cases where that happens.